_posts/2021-06-22-solidity-0.8.6-release-announcement.md

```diff
@@ -0,0 +1,60 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.6 Release Announcement'
+date: '2021-06-22'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.6](https://github.com/ethereum/solidity/releases/tag/v0.8.6) fixes some
+non-critical but annoying bugs, especially a warning about unreachable code that
+is in fact reachable.
+
+## Bugfixes
+
+### Unreachable Warning
+
+With Solidity 0.8.5, we released a new feature that detects if code is unreachable
+because a called function reverts unconditionally. This detection contained a bug
+where the "reverting behaviour" of a function was not properly updated after a
+function called by it was determined to be not always reverting.
+
+Because the bug was rather common, we are releasing this version with a rather
+short changelog.
+
+### Optimizer Setting
+
+We discovered that the default optimizer setting differed depending on the way
+the compiler was accessed: Using the compiler on the commandline ("``solc``")
+caused minimal optimizer features to be activated (the intended functionality)
+while using ``solc-js`` or Standard JSON all optimizer features were disabled.
+
+These minimal features are the "peephole optimizer", a routine that performs
+obvious replacements like removing ``PUSH1 0 POP`` and the "jumpdest remover"
+which removes code that is not jumped to from anywhere.
+
+The peephole optimizer and the jumpdest remover are activated even when
+setting ``settings.optimizer.enabled`` to ``false`` in Standard JSON
+and have to be explicitly disabled in the detailed settings.
+
+The bug was that omitting ``enabled`` was not equivalent to setting it to
+``false``: Omitting all optimizer settings was equivalent to explicitly
+disabling all optimizer components.
```

Comment:
Maybe we should clarify that it happened only when you omitted the `enabled` key but still had the `settings.optimizer` dict. If the whole dict was missing, the minimal settings were used.

The first paragraph of this section makes it sound as if it was the default Standard JSON behavior. Fortunately it was not so people who just omitted the whole dict were fine.

Reply:
Ah! I actually was not aware of that :)

Reply:
Sorry for that, I think I did not word that clearly enough in the changelog entry. I tried not to make it too long because I know you prefer them shorter but looks like I omitted too much important detail in this case.

Reply:
Added a fixup that rewords this section.

Reply:
Also, just in case I rechecked that behavior with 0.8.5 binary and I can confirm that omitting the whole dict works like setting `enabled` to `false`. Only providing a dict without `enabled` triggers this bug.